### PR TITLE
ryujinx: Add dependencies for the Avalonia UI

### PR DESCRIPTION
**Pull request by zhaofengli**
###### Description of changes

This PR adds required dependencies for the new Avalonia UI, and exposes the wrapped executable.

One issue that I ran into was that with `fcitx` enabled, SDL2 would segfault in `SDL_Fcitx_Init`. I worked around it with `SDL_IM_MODULE=none`.

###### Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [22.11 Release Notes (or backporting 22.05 Release notes)](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md#generating-2211-release-notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
  - [ ] (Release notes changes) Ran `nixos/doc/manual/md-to-db.sh` to update generated release notes
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by sbruder — APPROVED** [review]
Thanks! This works fine for me. Because I use sway, I had to run it with `GDK_BACKEND=x11 SDL_VIDEODRIVER=x11`, but that also is the case for the regular frontend.

**Comment by IvarWithoutBones** [comment]
> Thanks! This works fine for me. Because I use sway, I had to run it with `GDK_BACKEND=x11 SDL_VIDEODRIVER=x11`, but that also is the case for the regular frontend.

I think we should wrap the executables with those variables set by default. As far as I know that shouldn't case any regressions pn X11.

You can use `makeWrappersArgs` to append flags to the wrapper `buildDotnetModule` creates.

**Comment by zhaofengli** [comment]
I'll add the `GDK_BACKEND` variable, but I don't think forcing `SDL_IM_MODULE` is a good idea since it will break other IMEs like `ibus`.

**Comment by 06kellyjac** [comment]
In https://github.com/Ryujinx/Ryujinx/issues/2714 there's a lot of mentions of not touching the SDL variable but it seems pretty mixed

File: pkgs/applications/emulators/ryujinx/default.nix [hunk 1]
```diff
@@ -56,12 +64,27 @@ buildDotnetModule rec {
...
+    glew
   ];
 
   patches = [
     ./appdir.patch # Ryujinx attempts to write to the nix store. This patch redirects it to "~/.config/Ryujinx" on Linux.
   ];
 
+  makeWrapperArgs = [
+    "--set GDK_BACKEND x11"
+    "--set SDL_VIDEODRIVER x11"
```
Comment (IvarWithoutBones) [inline pkgs/applications/emulators/ryujinx/default.nix:85]: The `SDL_VIDEODRIVER=x11` flag doesn't make much sense considering SDL2 is never used for rendering. We only depend on `SDL2_mixer`, which is used for audio.
```suggestion
```

File: pkgs/applications/emulators/ryujinx/default.nix [hunk 2]
```diff
@@ -56,12 +64,27 @@ buildDotnetModule rec {
...
+    fontconfig
+    glew
   ];
 
   patches = [
     ./appdir.patch # Ryujinx attempts to write to the nix store. This patch redirects it to "~/.config/Ryujinx" on Linux.
   ];
 
+  makeWrapperArgs = [
+    "--set GDK_BACKEND x11"
```
Comment (IvarWithoutBones) [inline pkgs/applications/emulators/ryujinx/default.nix:84]:
```suggestion
    # Without this Ryujinx fails to start on wayland. See https://github.com/Ryujinx/Ryujinx/issues/2714
    "--set GDK_BACKEND x11"
```

**Review by IvarWithoutBones — CHANGES_REQUESTED** [review]

**Comment by IvarWithoutBones** [comment]
See https://github.com/Ryujinx/Ryujinx/pull/3516 as well, wayland might get native integration soon :)

**Review by IvarWithoutBones — APPROVED** [review]
The new UI works perfectly here, thanks for the keeping the derivation updated :)

Result of `nixpkgs-review pr 187070` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>ryujinx</li>
  </ul>
</details>

**Comment by IvarWithoutBones** [comment]
I noticed one potential problem with the avalonia UI:
```
00:00:00.024 |W| Application GetActualScaleFactor: Couldn't determine monitor DPI: Unrecognised XDG_SESSION_TYPE: tty
00:00:00.024 |W| Application GetActualScaleFactor: Couldn't determine monitor DPI: Unrecognised XDG_SESSION_TYPE: tty
```
If i recall correctly `tty` just refers to X11, kind of odd thats unknown. Launching it with `XDG_SESSION_TYPE=x11 ./results/ryujinx/bin/Ryujinx.Ava` makes the UI very unresponsive, although no warnings get logged.

DPI seemed fine anyways, I think its safe to ignore. Just thought I'd report it here.

File: pkgs/applications/emulators/ryujinx/default.nix (hunk 1 above)
Comment (zhaofengli) [reply pkgs/applications/emulators/ryujinx/default.nix]: Well, let's tack it back in #187150. Apparently SDL2 is used for input and is thus display protocol specific.

Comment (IvarWithoutBones) [reply pkgs/applications/emulators/ryujinx/default.nix]: Ah, my bad :sweat_smile:

**Pull request merged**